File: src/main/scala/com/github/julekarenalender/view/awt/util/LabelMaker.scala
package com.github.julekarenalender.view.awt.util

import no.jervell.view.awt.{Anchor, Scaling, Label}
import java.lang.String
import scala.Predef.String
import java.awt.{Color, Font}

object LabelPrinter {
  def apply() = new DefaultLabelMaker
}

class DefaultLabelMaker extends LabelMaker {
  private final val FontName: String = "Times New Roman"
  private final val FontSize: Int = 50
  private final val PlainFont: Font = new Font(FontName, Font.PLAIN, FontSize)
  private final val BoldFont: Font = new Font(FontName, Font.BOLD, FontSize)
  private final val TextColour: Color = Color.black

  def createDateLabel(date: Int): Label = {
    val label: Label = createLabel(String.valueOf(date))
    label.setFont(BoldFont)
    label
  }

  def createPersonLabel(name: String): Label = {
    val label: Label = createLabel(name.toUpperCase)
    label.setScaling(Scaling.STRETCH)
    label
  }

  def createLabel(text: String): Label = {
    val label: Label = new Label(text)
    label.setPaint(TextColour)
    label.setAnchor(Anchor.CENTER)
    label.setFont(PlainFont)
    label
  }
}

trait LabelMaker {
  def createPersonLabel(name: String): Label
  def createDateLabel(dateValue: Int): Label
  def createLabel(text: String): Label
}

File: src/main/kotlin/com/github/julekarenalender/gui/LabelMaker.kt
package com.github.julekarenalender.gui

import no.jervell.view.awt.Anchor
import no.jervell.view.awt.Scaling
import no.jervell.view.awt.Label
import java.awt.Color
import java.awt.Font

class LabelMaker {

    private val fontName: String = "Times New Roman"
    private val fontSize: Int = 50
    private val plainFont: Font = Font(fontName, Font.PLAIN, fontSize)
    private val boldFont: Font = Font(fontName, Font.BOLD, fontSize)
    private val textColour: Color = Color.black

    fun createDateLabel(date: Int): Label {
        val label: Label = createLabel(date.toString())
        label.font = boldFont
        return label
    }

    fun createPersonLabel(name: String): Label {
        val label: Label = createLabel(name.toUpperCase())
        label.scaling = Scaling.STRETCH
        return label
    }

    private fun createLabel(text: String): Label {
        val label = Label(text)
        label.paint = textColour
        label.anchor = Anchor.CENTER
        label.font = plainFont
        return label
    }
}